x-pack/plugins/enterprise_search/server/utils/elasticsearch_error_handler.ts

```diff
@@ -0,0 +1,73 @@
+/*
+ * Copyright Elasticsearch B.V. and/or licensed to Elasticsearch B.V. under one
+ * or more contributor license agreements. Licensed under the Elastic License
+ * 2.0; you may not use this file except in compliance with the Elastic License
+ * 2.0.
+ */
+
+import { RequestHandler } from '@kbn/core/server';
+import { Logger } from '@kbn/core/server';
+
+import { i18n } from '@kbn/i18n';
+
+import { ErrorCode } from '../../common/types/error_codes';
+
+import { createError, EnterpriseSearchError } from './create_error';
+import { isUnauthorizedException } from './identify_exceptions';
+
+export function elasticsearchErrorHandler<ContextType, RequestType, ResponseType>(
+  log: Logger,
+  requestHandler: RequestHandler<ContextType, RequestType, ResponseType>
+): RequestHandler<ContextType, RequestType, ResponseType> {
+  return async (context, request, response) => {
```

Comment:
Do the context/request/response params here not need to be typed? You have the generic types anyway.

Reply:
No, they get their typings from `RequestHandler<>`

<img width="797" alt="image" src="https://user-images.githubusercontent.com/2479295/183895215-a6c0bf89-0670-4279-b0e3-8655128542c5.png">